services/thermal_mgr/thermal_mgr.c

```diff
@@ -44,18 +44,70 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
-  return ERR_CODE_SUCCESS;
+  if (thermalMgrQueueHandle == NULL || event == NULL)
+  {
+    return ERR_CODE_INVALID_ARG; 
+  }
+
+  if (xQueueSend(thermalMgrQueueHandle, (void *) event, 0) == pdTRUE)
+  {
+    return ERR_CODE_SUCCESS;
+  }  
+
+  return ERR_CODE_QUEUE_FULL; 
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t interrupt;
+  interrupt.type = OVER_TEMPERATURE_SHUTDOWN;
+  thermalMgrSendEvent(&interrupt);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  thermal_mgr_event_t event;
+   
+  lm75bd_config_t data = *(lm75bd_config_t *) pvParameters;
+
+  
   while (1) {
-    
+    if (thermalMgrQueueHandle != NULL)
+    {
+    if (xQueueReceive(thermalMgrQueueHandle, &event, 5000/portTICK_PERIOD_MS) == pdPASS)
+    {
+        float currtemp;
+        if (event.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD)
+        {
+          error_code_t errCode = readTempLM75BD(data.devAddr, &currtemp);
```

Comment:
We usually want to also log the error code. We tend to use the LOG_IF_ERROR_CODE macro in logging.h